website/src/routes/routing.ts

```diff
@@ -9,61 +9,73 @@ import { WestNileAnalyzeSingleVariantView, WestNileSequencingEffortsView } from
-                new H5n1SequencingEffortsView(organismsConfig),
-            ],
-            [Organisms.mpox]: [
-                new MpoxAnalyzeSingleVariantView(organismsConfig),
-                new MpoxSequencingEffortsView(organismsConfig),
-            ],
-            [Organisms.rsvA]: [
-                new RsvAAnalyzeSingleVariantView(organismsConfig),
-                new RsvASequencingEffortsView(organismsConfig),
-            ],
-            [Organisms.rsvB]: [
-                new RsvBAnalyzeSingleVariantView(organismsConfig),
-                new RsvBSequencingEffortsView(organismsConfig),
-            ],
-            [Organisms.westNile]: [
-                new WestNileAnalyzeSingleVariantView(organismsConfig),
-                new WestNileSequencingEffortsView(organismsConfig),
-            ],
+            [Organisms.covid]: {
+                [singleVariantViewKey]: new CovidAnalyzeSingleVariantView(organismsConfig),
+                [compareVariantsViewKey]: new CovidCompareVariantsView(
+                    organismsConfig,
+                    loggerProvider('CovidCompareVariantsView'),
+                ),
+                [sequencingEffortsViewKey]: new CovidSequencingEffortsView(organismsConfig),
+            },
+            [Organisms.h5n1]: {
+                [singleVariantViewKey]: new H5n1AnalyzeSingleVariantView(organismsConfig),
+                [sequencingEffortsViewKey]: new H5n1SequencingEffortsView(organismsConfig),
+            },
+            [Organisms.mpox]: {
+                [singleVariantViewKey]: new MpoxAnalyzeSingleVariantView(organismsConfig),
+                [sequencingEffortsViewKey]: new MpoxSequencingEffortsView(organismsConfig),
+            },
+            [Organisms.rsvA]: {
+                [singleVariantViewKey]: new RsvAAnalyzeSingleVariantView(organismsConfig),
+                [sequencingEffortsViewKey]: new RsvASequencingEffortsView(organismsConfig),
+            },
+            [Organisms.rsvB]: {
+                [singleVariantViewKey]: new RsvBAnalyzeSingleVariantView(organismsConfig),
+                [sequencingEffortsViewKey]: new RsvBSequencingEffortsView(organismsConfig),
+            },
+            [Organisms.westNile]: {
+                [singleVariantViewKey]: new WestNileAnalyzeSingleVariantView(organismsConfig),
+                [sequencingEffortsViewKey]: new WestNileSequencingEffortsView(organismsConfig),
+            },
         } as const;
-        this.allViews = Object.values(this.views).flat();
-    }
-
-    public get covidAnalyzeSingleVariantView() {
-        return this.views.covid[0];
     }
 
     public get covidCompareVariantsView() {
-        return this.views.covid[1];
+        return this.views.covid[compareVariantsViewKey];
     }
 
-    public getCurrentRouteInBrowser = () => {
-        return this.parseUrl(new URL(window.location.href));
-    };
+    public getOrganismView<Organism extends keyof ViewsMap, Key extends ViewKey<Organism>>(
```

Comment:
Those are overloads. Apparentely that's how it works in Typescript: You define the method with different signature. Then you have one implementation that must match all previously defined signatures. Otherwise I couldn't get rid of all TS errors - essentially the two versions mirror the two use cases:
* You pass a concrete key and want to get back the concrete type of the `View`.
* You get a key of which you don't know the exact type, just that it conforms to the OrganismViewKey union type. Then you only get back a generic view, without further specification.